typhos/cli.py

```diff
@@ -107,15 +178,162 @@ def _create_happi_client(cfg):
+        Options for the scrollbar. See the cli help for valid options.
+
+    Returns
+    -------
+    suite : TyphosSuite
+        A suite that has been populated with devices.
+    """
     if device_names:
-        devices = create_devices(device_names, cfg=cfg,
-                                 fake_devices=fake_devices)
+        devices = create_devices(
+            device_names, cfg=cfg, fake_devices=fake_devices
+        )
     else:
         devices = []
     if devices or not device_names:
-        return typhos.TyphosSuite.from_devices(devices)
+        layout_obj = get_layout_from_cli(layout, cols)
+        display_type_enum = get_display_type_from_cli(display_type)
+        scroll_enum = get_scrollable_from_cli(scroll_option)
+        return typhos.TyphosSuite.from_devices(
+            devices,
+            content_layout=layout_obj,
+            default_display_type=display_type_enum,
+            scroll_option=scroll_enum,
+        )
+
+
+def get_layout_from_cli(
+    layout: str,
+    cols: int,
+) -> QtWidgets.QLayout:
+    """
+    Return a correct layout object based on user input.
+
+    Parameters
+    ----------
+    layout : str
+        String representation of the layout.
+    cols : int
+        Number of columns to use for the grid layout.
+
+    Returns
+    -------
+    qlayout : QLayout
+        The qt layout object, instantiated.
+    """
+    if 'horizontal'.startswith(layout):
+        return QtWidgets.QHBoxLayout()
+    if 'vertical'.startswith(layout):
+        return QtWidgets.QVBoxLayout()
+    if 'grid'.startswith(layout):
+        return FixedColGrid(cols=cols)
+    if 'flow'.startswith(layout):
+        return FlowLayout()
+    else:
+        raise ValueError(
+            f'{layout} is not a valid layout name. '
+            'The allowed values are "horizontal", '
+            '"vertical", "grid", and "flow".'
+        )
+
+
+class FixedColGrid(QtWidgets.QGridLayout):
+    """
+    A QGridLayout with fixed number of columns.
+
+    QGridLayout allows us to add widgets at any row or column, but
+    for the purposes of typhos we'd like to be able to pass in just
+    the widget like for other layouts. As such, we select a fixed
+    number of columns and fill devices in row-by-row, left-to-right
+    first and then top-to-bottom.
+    """
+
+    def __init__(
+        self,
+        *args,
+        cols: int = 3,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self._n_cols = cols
+        self._widget_index = 0
+
+    def addWidget(
+        self,
+        widget: QtWidgets.QWidget,
+        *args,
+    ):
+        row = self._widget_index // self._n_cols
+        col = self._widget_index % self._n_cols
+        super().addWidget(widget, row, col, *args)
+        self._widget_index += 1
+
+
+def get_display_type_from_cli(display_type: str) -> DisplayTypes:
+    """Convert the cli string to the appropriate DisplayTypes enum."""
+    if 'embedded'.startswith(display_type):
```

Comment:
Would be kinda nice to do this directly on the enum, but then again it probably won't be reused outside of the CLI, so :meh:

Reply:
Is it worth putting an enum base in pcdsutils for this? This should be achievable generically...

Reply:
It's generally useful for command-line input, so maybe?